### PR TITLE
refactor: use minimatch util

### PR DESCRIPTION
**Pull request by setchy**
## Changes

refactor minimatch match use-cases to use cacheable util

## Context

* https://github.com/renovatebot/renovate/issues/23491

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/extract/utils.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import { minimatch } from 'minimatch';
 import { logger } from '../../../../logger';
+import { minimatch } from '../../../../util/minimatch';
 
 export function matchesAnyPattern(val: string, patterns: string[]): boolean {
   const res = patterns.some(
-    (pattern) => pattern === val + '/' || minimatch(val, pattern, { dot: true })
+    (pattern) =>
+      pattern === val + '/' || minimatch(pattern, { dot: true }).match(val)
```
Comment (viceice) [inline lib/modules/manager/npm/extract/utils.ts:7]: convert first part to string template

**Review by viceice — COMMENTED** [review]
can you measure any performance improvement?

**Comment by setchy** [comment]
> can you measure any performance improvement?

I haven't spent any time on measuring any performance gains

File: lib/modules/manager/npm/extract/utils.ts (hunk 1 above)
Comment (setchy) [reply lib/modules/manager/npm/extract/utils.ts]: updated

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**